src/signing/mod.rs

```diff
@@ -74,10 +76,11 @@ impl SignerHandle {
 
         Ok(match signer_type {
             #[cfg(feature = "stronghold")]
-            SignerTypeDto::Stronghold(stronghold_dto) => StrongholdSigner::try_new_signer_handle(
-                &stronghold_dto.password,
-                Path::new(&stronghold_dto.snapshot_path),
-            )?,
+            SignerTypeDto::Stronghold(stronghold_dto) => StrongholdSigner::builder()
+                .password(&stronghold_dto.password)
```

Comment:
Please make the password in the dto optional and only set it here if it's `Some`